lib/awful/tooltip.lua

```diff
@@ -189,14 +483,28 @@ end
 --   seconds.
 -- @tparam[opt=apply_dpi(5)] integer args.margin_leftright The left/right margin for the text.
 -- @tparam[opt=apply_dpi(3)] integer args.margin_topbottom The top/bottom margin for the text.
+-- @tparam[opt=nil] gears.shape args.shape The shape
 -- @treturn awful.tooltip The created tooltip.
 -- @see add_to_object
--- @see set_timeout
--- @see set_text
--- @see set_markup
-tooltip.new = function(args)
-    local self = setmetatable(object(), instance_mt)
-    self.visible = false
+-- @see timeout
+-- @see text
+-- @see markup
+-- @function awful.tooltip
+function tooltip.new(args)
+    local self = object {
+        enable_properties = true,
+    }
+
+    rawset(self,"_private", {})
+
+    self:add_signal("property::visible")
+    self:add_signal("property::align")
+    self:add_signal("property::mode")
```

Comment:
I haven't looked at this at all. I just saw a metric ton of warnings in the travis build (the LuaJIT-build failed because of `tests/test-screen-changes.lua`, our unstable test, running into a timeout).
Since commit 0857f6f1b5535a862c840311f5bdad849f42e95d, `add_signal` doesn't do anything except for printing a warning, so these three calls can be removed.